clients/src/main/java/org/apache/kafka/common/requests/CreatePartitionsRequest.java

```diff
@@ -72,17 +71,47 @@
     // It is an error for duplicate topics to be present in the request,
     // so track duplicates here to allow KafkaApis to report per-topic errors.
     private final Set<String> duplicates;
-    private final Map<String, NewPartitions> newPartitions;
+    private final Map<String, PartitionDetails> newPartitions;
     private final int timeout;
     private final boolean validateOnly;
 
+    public static class PartitionDetails {
+
+        private final int totalCount;
+
+        private final List<List<Integer>> newAssignments;
```

Comment:
Should this be simply `assignments` for consistency with the accessor name? Same for the construction parameter. Or change the accessor name.

Reply:
@ijuma Thanks for the review. I changed the accessor name to make it consistent with the field name. Since the count is a total and assignments are delta, it feels better to keep the `new` prefix for the field and accessor.